pkg/networkpolicy/networkpolicy.go

```diff
@@ -0,0 +1,321 @@
+package networkpolicy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	netv1 "k8s.io/api/networking/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NetworkPolicyBuilder provides struct for networkPolicy object.
+type NetworkPolicyBuilder struct {
+	// NetworkPolicy definition. Used to create networkPolicy object with minimum set of required elements.
+	Definition *netv1.NetworkPolicy
+	// Created networkPolicy object on the cluster.
+	Object *netv1.NetworkPolicy
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before NetworkPolicy object is created.
+	errorMsg string
+}
+
+// NewNetworkPolicyBuilder method creates new instance of builder.
+func NewNetworkPolicyBuilder(apiClient *clients.Settings, name, nsname string) *NetworkPolicyBuilder {
+	glog.V(100).Infof("Initializing new NetworkPolicyBuilder structure with the following params: name: %s, namespace: %s",
+		name, nsname)
+
+	builder := &NetworkPolicyBuilder{
+		apiClient: apiClient,
+		Definition: &netv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the networkPolicy is empty")
+
+		builder.errorMsg = "The networkPolicy 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the networkPolicy is empty")
+
+		builder.errorMsg = "The networkPolicy 'namespace' cannot be empty"
+	}
+
+	return builder
+}
+
+// WithNamespaceIngressRule applies ingress rule for the networkPolicy.
+func (builder *NetworkPolicyBuilder) WithNamespaceIngressRule(
+	namespaceIngressMatchLabels map[string]string,
+	podIngressMatchLabels map[string]string) *NetworkPolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
```

Comment:
We usually just return err here. builder.errorMsg will already be set in the validate method if something goes wrong

Reply:
It's duplication since we have exactly the same logic on line 65

Reply:
Right, line 65 should be removed and the return should just be done here. That's how validate is used in the other methods

Reply:
Ok I got it.
```
if valid, err := builder.validate(); !valid {
	builder.errorMsg = err.Error()

        return bulder
}

```
And get rid of next condition on line 65

Reply:
I could not return err here since only the object I return from this method is NetworkPolicyBuilder builder

Reply:
ok, now I see what you mean